python/pylibcudf/pylibcudf/column.pyx

```diff
@@ -154,6 +155,44 @@ class _Ravelled:
         self.__cuda_array_interface__ = cai
 
 
+def _prepare_array_metadata(
+    iface: dict
+) -> tuple[int, int, tuple[int, ...], tuple[int, ...] | None, DataType]:
+    """
+    Parse and validate a CUDA or NumPy array interface dict.
+    """
+    typestr = iface["typestr"]
+    shape = iface["shape"]
+    strides = iface.get("strides")
+    data = iface.get("data")
+
+    if typestr[0] == ">":
+        raise ValueError("Big-endian data is not supported")
+    if not isinstance(data, tuple) or not isinstance(data[0], int):
+        raise ValueError(
+            "Expected a data field with an integer pointer in the array interface. "
+            "Objects with data set to None or a buffer object are not supported."
+        )
+    if not isinstance(shape, tuple) or len(shape) == 0:
+        raise ValueError("shape must be a non-empty tuple")
+    if len(shape) > 2:
+        raise ValueError("Only 1D or 2D arrays are supported")
+    dtype = _datatype_from_dtype_desc(typestr[1:])
+    itemsize = size_of(dtype)
+    if not is_c_contiguous(shape, strides, itemsize):
+        raise ValueError("Data must be C-contiguous")
+    if shape[0] >= numeric_limits[size_type].max():
+        raise ValueError(
+            "Number of rows exceeds size_type limit for offsets column construction."
+        )
```

Comment:
Note: (let's fix this in a followup). If `len(shape) == 1` (so not a list column construction) this is off by one (because we just need that `shape[0] >`, not `>=`).

Reply:
Thanks, I'll tack that on to my PR supporting general ndarrays.